### PR TITLE
TestInitListeners: Use port 0 pick available port

### PR DESCRIPTION
**Pull request by krisis**
## Description

With this change TestInitListeners picks port "0" to avoid failing with "Port already in use".

## Motivation and Context

See #4505 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/server-mux_test.go [hunk 1]
```diff
@@ -109,20 +109,18 @@ func dial(addr string) error {
 
 // Tests initializing listeners.
 func TestInitListeners(t *testing.T) {
-	portTest1 := getFreePort()
-	portTest2 := getFreePort()
```
Comment (harshavardhana) [inline cmd/server-mux_test.go:113]: GetFreePort uses port '0' to get the free port. GetFreePort should work appropriately in the first place did you investigate why it doesn't work ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/server-mux_test.go (hunk 1 above)
Comment (krisis) [reply cmd/server-mux_test.go:113]: @harshavardhana I am aware of how getFreePort works. It is obvious from the travis-ci failure that the port returned by getFreePort wasn't really free. getFreePort closes the listener (and the underlying socket) leaving the port free for other applications to bind to. So, getFreePort doesn't guarantee that the port it finds free to be available indefinitely. This is why the unit test failed. Whether port will remain free during the test is non-deterministic and can't be relied on. Does this satisfactorily explain the problem?

Comment (harshavardhana) [reply cmd/server-mux_test.go:113]: Yes now it makes sense.. just wanted to see why its racy.

**Review by harshavardhana — APPROVED** [review]

File: cmd/server-mux_test.go (hunk 1 above)
Comment (donatello) [reply cmd/server-mux_test.go:113]: Yes, `getFreePort` is inherently racy and should not be implemented like this. The right way is to write a function that creates a socket bound to a free port and return that. In the meanwhile lets get this change in.

**Review by donatello — APPROVED** [review]
LGTM

**Pull request merged**